cli/__snapshots__/cli_spec.js

```diff
@@ -168,3 +168,30 @@ exports['cli -v no binary version 1'] = `
 Cypress package version: 1.2.3
 Cypress binary version: not installed
 `
+
+exports['cli CYPRESS_ENV allows staging environment 1'] = `
+  code: 0
+  stderr:
+  -------
+  
+  -------
+
+`
+
+exports['cli CYPRESS_ENV catches environment "foo" 1'] = `
+  code: 11
+  stderr:
+  -------
+  We have detected unknown or unsupported CYPRESS_ENV value
```

Comment:
Taking a shot at this - I know this makes it less descriptive for 'internal' dev, but maybe more descriptive to users.

```
Reserved word "CYPRESS_ENV" was used.

Remove "CYPRESS_ENV" variable and run Cypress again.
```

Reply:
I would change it a little bit because "reserved word" does not really describe it
```
Environment variable with reserved name "CYPRESS_ENV" was used.

Remove "CYPRESS_ENV" variable and run Cypress again.
```

Reply:
@bahmutov ^ I like yours best, especially specifying that it is an environment variable.